apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -232,6 +232,17 @@ defmodule Astarte.AppEngine.API.Device do
            Queries.retrieve_mapping(client, interface_descriptor.interface_id, endpoint_id),
          {:ok, value} <- cast_value(mapping.value_type, raw_value),
          :ok <- validate_value_type(mapping.value_type, value),
+         wrapped_value = wrap_to_bson_struct(mapping.value_type, value),
+         publish_opts = build_publish_opts(interface_descriptor.type, mapping.reliability),
```

Comment:
if you don't mind we should pattern match type, reliability and name so we can write shorter lines